spec/ios_l10n_helper_spec.rb

```diff
@@ -0,0 +1,162 @@
+require_relative './spec_helper'
+
+describe Fastlane::Helper::Ios::L10nHelper do
+  let(:test_data_dir) { File.join(File.dirname(__FILE__), 'test-data', 'translations', 'ios_l10n_helper') }
+
+  def fixture(name)
+    File.join(test_data_dir, name)
+  end
+
+  # Returns the `Encoding` instance guessed for that file. Useful to control that our fixtures are in the expected encoding when running tests.
+  def file_encoding(path)
+    File.read(path, mode: 'rb:BOM|UTF-8').encoding # Use BOM to determine encoding if present, fallback to UTF8
+  end
+
+  describe '#strings_file_type' do
+    it 'detects an XML-formatted strings file' do
+      xml_fixture = fixture('xml-format.strings')
+      expect(described_class.strings_file_type(path: xml_fixture)).to eq(:xml)
+    end
+
+    it 'detects a binary-plist strings file' do
+      bplist_fixture = fixture('bplist-format.strings')
+      expect(described_class.strings_file_type(path: bplist_fixture)).to eq(:binary)
+    end
+
+    it 'detects a text-formatted strings file' do
+      text_fixture = fixture('Localizable-utf16.strings')
+      expect(described_class.strings_file_type(path: text_fixture)).to eq(:text)
+    end
+
+    it 'returns nil on a non-existing path' do
+      expect(described_class.strings_file_type(path: '/invalid-path')).to be_nil
+    end
+
+    it 'returns nil on file that is not a strings file at all' do
+      png_fixture = fixture('not-a-strings-file.png')
```

Comment:
Is there a particular reason you used a PNG as opposed to a text file that is not a `.strings`?

Clearly, an image is quite different from a `.strings` file and therefore represents a good test input. I just "worry" about tracking images in the repo when we don't have to. Git is good at managing text files, not images/binaries.